From cecab8373b68ba8b726c5e654bb903bbb6580181 Mon Sep 17 00:00:00 2001
From: Liming Wang <liming.wang@windriver.com>
Date: Tue, 25 Mar 2014 11:40:39 +0800
Subject: [PATCH 070/628] mmc: host: xilinxps: __devinit, __devexit,
 __definitdata, __devexit_p, __exit_p

git://github.com/Xilinx/linux-xlnx.git xilinx-v14.7
commit 068160c0 Merge commit 'v3.8' into master-next

cherry picked the sdhci driver modification.

Signed-off-by: Liming Wang <liming.wang@windriver.com>
---
 drivers/mmc/host/sdhci-of-xilinxps.c |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/mmc/host/sdhci-of-xilinxps.c b/drivers/mmc/host/sdhci-of-xilinxps.c
index 215093e..f79e0db 100644
--- a/drivers/mmc/host/sdhci-of-xilinxps.c
+++ b/drivers/mmc/host/sdhci-of-xilinxps.c
@@ -140,7 +140,7 @@ static const struct dev_pm_ops xsdhcips_dev_pm_ops = {
 #define XSDHCIPS_PM	NULL
 #endif /* ! CONFIG_PM_SLEEP */
 
-static int __devinit sdhci_zynq_probe(struct platform_device *pdev)
+static int sdhci_zynq_probe(struct platform_device *pdev)
 {
 	int ret;
 	int irq = platform_get_irq(pdev, 0);
@@ -229,7 +229,7 @@ err_free:
 	return ret;
 }
 
-static int __devexit sdhci_zynq_remove(struct platform_device *pdev)
+static int sdhci_zynq_remove(struct platform_device *pdev)
 {
 	struct sdhci_host *host = platform_get_drvdata(pdev);
 	struct sdhci_pltfm_host *pltfm_host = sdhci_priv(host);
@@ -246,7 +246,7 @@ static int __devexit sdhci_zynq_remove(struct platform_device *pdev)
 	return sdhci_pltfm_unregister(pdev);
 }
 
-static const struct of_device_id sdhci_zynq_of_match[] __devinitconst = {
+static const struct of_device_id sdhci_zynq_of_match[] = {
 	{ .compatible = "xlnx,ps7-sdhci-1.00.a" },
 	{ .compatible = "generic-sdhci" },
 	{},
@@ -261,7 +261,7 @@ static struct platform_driver sdhci_zynq_driver = {
 		.pm = XSDHCIPS_PM,
 	},
 	.probe = sdhci_zynq_probe,
-	.remove = __devexit_p(sdhci_zynq_remove),
+	.remove = sdhci_zynq_remove,
 };
 
 module_platform_driver(sdhci_zynq_driver);
-- 
1.7.5.4

